crates/host/src/wasmbus/mod.rs

```diff
@@ -2998,8 +2998,22 @@ impl Host {
                                                 warn!(?e, "failed to publish provider health check failed event");
                                             }
                                         }
```

Comment:
Can't comment on the appropriate line since it's up there quite a ways, but can the child `spawn` be its own function? This function is quite long

Reply:
There is a lot to refactor here, but I don't think this change belongs in this PR